gitclone/gitclone_test.go

```diff
@@ -12,608 +12,565 @@ import (
-			fetchFailedTag,
-			fmt.Errorf("fetch failed: %v", errors.New(rawCmdError)),
-			"Fetching repository has failed",
-			"fake",
-			[]string{"master"},
-		),
-	},
-	{
-		name: "PR - no fork - auto merge: BranchDest missing (UNSUPPORTED)",
-		cfg: Config{
-			Commit:        "76a934ae",
-			Branch:        "test/commit-messages",
-			PRMergeBranch: "pull/7/merge",
-			CloneDepth:    1,
-			ShouldMergePR: true,
-		},
-		wantErrType: ParameterValidationError{},
-		wantCmds:    nil,
-	},
 
-	// ** CloneDepth specified, Unshallow needed **
-	{
-		name: "Checkout commit, unshallow needed",
-		cfg: Config{
-			Commit:           "cfba2b01332e31cb1568dbf3f22edce063118bae",
-			CloneDepth:       1,
-			UpdateSubmodules: true,
-		},
-		mockRunner: givenMockRunner().
-			GivenRunFailsForCommand(`git "checkout" "cfba2b01332e31cb1568dbf3f22edce063118bae"`, 1).
-			GivenRunWithRetrySucceeds().
-			GivenRunSucceeds(),
-		wantCmds: []string{
-			`git "fetch" "--jobs=10" "--depth=1" "--no-tags"`,
-			`git "checkout" "cfba2b01332e31cb1568dbf3f22edce063118bae"`,
-			// fatal: reference is not a tree: cfba2b01332e31cb1568dbf3f22edce063118bae
-			// Checkout failed, error: fatal: reference is not a tree: cfba2b01332e31cb1568dbf3f22edce063118bae
-			`git "fetch" "--jobs=10" "--unshallow" "--no-tags"`,
-			`git "checkout" "cfba2b01332e31cb1568dbf3f22edce063118bae"`,
+		// ** Errors **
+		{
+			name: "Checkout nonexistent branch",
+			cfg: Config{
+				Branch: "fake",
+			},
+			mockRunner: givenMockRunner().
+				GivenRunWithRetryFailsAfter(2).
+				GivenRunSucceeds(),
+			wantCmds: []string{
+				`git "fetch" "--jobs=10" "--no-tags" "--no-recurse-submodules" "origin" "refs/heads/fake"`,
+				`git "fetch" "--jobs=10" "--no-tags" "--no-recurse-submodules" "origin" "refs/heads/fake"`,
+				`git "fetch" "--jobs=10" "--no-tags" "--no-recurse-submodules" "origin" "refs/heads/fake"`,
+				`git "fetch" "--jobs=10"`,
+				`git "branch" "-r"`,
+			},
+			wantErr: newStepErrorWithBranchRecommendations(
+				fetchFailedTag,
+				fmt.Errorf("fetch failed: %v", errors.New(rawCmdError)),
+				"Fetching repository has failed",
+				"fake",
+				[]string{"master"},
+			),
+		},
+		{
+			name: "PR - no fork - auto merge: BranchDest missing (UNSUPPORTED)",
+			cfg: Config{
+				Commit:        "76a934ae",
+				Branch:        "test/commit-messages",
+				PRMergeBranch: "pull/7/merge",
+				CloneDepth:    1,
+				ShouldMergePR: true,
+			},
+			wantErrType: ParameterValidationError{},
+			wantCmds:    nil,
 		},
-	},
-	{
-		name: "PR - no fork - manual merge: branch, no commit (ignore depth) UNSUPPORTED?",
-		cfg: Config{
-			Branch:        "test/commit-messages",
-			PRMergeBranch: "pull/7/merge",
-			PRDestBranch:  "master",
-			ManualMerge:   true,
-			ShouldMergePR: true,
-		},
-		wantErrType: ParameterValidationError{},
-		wantCmds:    nil,
-	},
-	{
-		name: "Checkout PR - auto merge - merge branch, with depth (unshallow needed)",
-		cfg: Config{
```

Comment:
This test case is now deleted. It doesn't make sense, with the fixed checkout strategy we don't do a merge anymore, so the shallow fetch is not a problem.